src/renate/memory/storage.py

```diff
@@ -0,0 +1,151 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+import math
+import os
+from typing import Tuple, Union
+
+import torch
+from renate.types import NestedTensors
+
+Index = Union[str, slice]
+
+TORCH_DTYPES_BY_STRING = {
+    "torch.float32": torch.float32,
+    "torch.float64": torch.float64,
+    "torch.int32": torch.int32,
+    "torch.int64": torch.int64,
+}
+
+
+def mmap_tensor(filename: str, shape: Tuple[int, ...], dtype: torch.dtype) -> torch.Tensor:
+    """Creates or accesses a memory-mapped tensor."""
+    t = torch.from_file(filename, shared=True, size=math.prod(shape), dtype=dtype, device="cpu")
+    return t.view(shape)
+
+
+def _create_or_access_storage(path: str, length: int, shapes, dtypes) -> NestedTensors:
```

Comment:
There would be an alternative way to implement this, where instead of giving `shapes` and `dtypes`, we just give a single prototype data point. That would also make superfluous the `get_nested_tensor_schema` method and the need to convert dtypes to strings, et cetera. If you'd like that solution better, I could replace it.

Reply:
@wistuba Thoughts on this?

Reply:
I decided to go with this, see latest revision.